In [2]:
import pandas as pd
import numpy as np

from google.colab import files
uploaded = files.upload()

df = pd.read_csv("Day16_Student_Wellbeing_Survey.csv")

df.head()

,Student_ID,Age,Faculty,Year_of_Study,City,Accommodation,Scholarship,Part_Time_Job,Internet_Quality,Preferred_Study_Space,Weekly_Study_Hours,Average_Sleep_Hours,Daily_Screen_Time_Hours,Exercise_Days_Per_Week,Commute_Time_Minutes,Stress_Score,Academic_Readiness_Score,Overall_Satisfaction,Monthly_Discretionary_Spending,Social_Activity_Hours_Per_Week
0,STU0001,21,Data Science,3,Bengaluru,Hostel,No,Yes,Good,Café,15.2,5.7,3.2,3,11.8,5.4,63.1,3.2,5131.0,7.8
1,STU0002,21,Data Science,4,Pune,Hostel,Yes,No,Poor,Library,17.6,7.7,3.5,4,20.3,2.1,74.2,3.8,3508.0,9.2
2,STU0003,22,Life Sciences,2,Chandigarh,Home,Yes,No,Poor,Home,18.9,6.4,2.8,1,15.0,5.2,77.3,4.0,4584.0,7.7
3,STU0004,18,Business,4,Bengaluru,Hostel,Yes,Yes,Average,Café,19.5,6.7,3.0,5,12.1,4.7,67.0,4.0,5456.0,5.3
4,STU0005,20,Business,3,Jammu,Shared Apartment,Yes,No,Good,Home,18.4,7.3,3.0,1,30.2,5.4,65.6,3.2,11732.0,15.9


In [3]:
print(df.info())
print(df.describe())
print(df.isnull().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 600 entries, 0 to 599
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Student_ID                      600 non-null    object 
 1   Age                             600 non-null    int64  
 2   Faculty                         600 non-null    object 
 3   Year_of_Study                   600 non-null    int64  
 4   City                            600 non-null    object 
 5   Accommodation                   600 non-null    object 
 6   Scholarship                     600 non-null    object 
 7   Part_Time_Job                   600 non-null    object 
 8   Internet_Quality                600 non-null    object 
 9   Preferred_Study_Space           600 non-null    object 
 10  Weekly_Study_Hours              600 non-null    float64
 11  Average_Sleep_Hours             600 non-null    float64
 12  Daily_Screen_Time_Hours         600 

In [4]:
columns = ["Weekly_Study_Hours","Average_Sleep_Hours",
           "Daily_Screen_Time_Hours","Stress_Score",
           "Academic_Readiness_Score"]

for col in columns:
    print("\n", col)
    print("Mean:", df[col].mean())
    print("Median:", df[col].median())
    print("Mode:", df[col].mode()[0])
    print("Range:", df[col].max() - df[col].min())
    print("Variance:", df[col].var())
    print("Standard Deviation:", df[col].std())


 Weekly_Study_Hours
Mean: 15.690999999999997
Median: 15.4
Mode: 13.1
Range: 30.0
Variance: 19.217047412353942
Standard Deviation: 4.383725289334853

 Average_Sleep_Hours
Mean: 6.9975
Median: 7.0
Mode: 7.0
Range: 5.0
Variance: 0.7033827212020047
Standard Deviation: 0.8386791527169403

 Daily_Screen_Time_Hours
Mean: 4.503
Median: 4.2
Mode: 3.5
Range: 11.200000000000001
Variance: 3.469006010016694
Standard Deviation: 1.8625267810199921

 Stress_Score
Mean: 4.4645
Median: 4.5
Mode: 4.7
Range: 8.4
Variance: 2.7824939065108416
Standard Deviation: 1.6680809052653416

 Academic_Readiness_Score
Mean: 71.77316666666667
Median: 71.65
Mode: 71.1
Range: 56.1
Variance: 93.98196658319426
Standard Deviation: 9.694429667762527


In [5]:
for col in columns:
    q1 = df[col].quantile(0.25)
    q3 = df[col].quantile(0.75)
    iqr = q3 - q1

    print("\n", col)
    print("Q1:", q1)
    print("Q3:", q3)
    print("IQR:", iqr)


 Weekly_Study_Hours
Q1: 13.0
Q3: 18.424999999999997
IQR: 5.424999999999997

 Average_Sleep_Hours
Q1: 6.475
Q3: 7.6
IQR: 1.125

 Daily_Screen_Time_Hours
Q1: 3.2
Q3: 5.4
IQR: 2.2

 Stress_Score
Q1: 3.3
Q3: 5.7
IQR: 2.4000000000000004

 Academic_Readiness_Score
Q1: 65.2
Q3: 78.025
IQR: 12.825000000000003


In [6]:
std_values = df[columns].std()
print(std_values)

print("Highest Variability:", std_values.idxmax())

Weekly_Study_Hours          4.383725
Average_Sleep_Hours         0.838679
Daily_Screen_Time_Hours     1.862527
Stress_Score                1.668081
Academic_Readiness_Score    9.694430
dtype: float64
Highest Variability: Academic_Readiness_Score


In [7]:
outlier_columns = ["Weekly_Study_Hours",
                   "Daily_Screen_Time_Hours",
                   "Commute_Time_Minutes",
                   "Monthly_Discretionary_Spending"]

for col in outlier_columns:
    q1 = df[col].quantile(0.25)
    q3 = df[col].quantile(0.75)
    iqr = q3 - q1

    lower = q1 - 1.5*iqr
    upper = q3 + 1.5*iqr

    outliers = df[(df[col] < lower) | (df[col] > upper)]

    print(col, "Outliers:", len(outliers))

Weekly_Study_Hours Outliers: 8
Daily_Screen_Time_Hours Outliers: 18
Commute_Time_Minutes Outliers: 4
Monthly_Discretionary_Spending Outliers: 20


In [8]:
col = "Daily_Screen_Time_Hours"

q1 = df[col].quantile(0.25)
q3 = df[col].quantile(0.75)
iqr = q3 - q1

lower = q1 - 1.5*iqr
upper = q3 + 1.5*iqr

clean_df = df[(df[col] >= lower) & (df[col] <= upper)]

print("Mean Before:", df[col].mean())
print("Mean After:", clean_df[col].mean())

print("Median Before:", df[col].median())
print("Median After:", clean_df[col].median())

Mean Before: 4.503
Mean After: 4.31340206185567
Median Before: 4.2
Median After: 4.1


In [9]:
A = len(df[df["Part_Time_Job"]=="Yes"]) / len(df)
B = len(df[df["Stress_Score"]>7]) / len(df)
C = len(df[df["Scholarship"]=="Yes"]) / len(df)
D = len(df[df["Exercise_Days_Per_Week"]>=3]) / len(df)

AB = len(df[(df["Part_Time_Job"]=="Yes") &
            (df["Stress_Score"]>7)]) / len(df)

AC = len(df[(df["Part_Time_Job"]=="Yes") &
            (df["Scholarship"]=="Yes")]) / len(df)

PA_given_B = AB / B
PB_given_A = AB / A

print("P(A):", A)
print("P(B):", B)
print("P(C):", C)
print("P(D):", D)
print("P(A∩B):", AB)
print("P(A∩C):", AC)
print("P(A|B):", PA_given_B)
print("P(B|A):", PB_given_A)
print("P(A)×P(B):", A*B)

P(A): 0.25333333333333335
P(B): 0.06666666666666667
P(C): 0.30666666666666664
P(D): 0.5933333333333334
P(A∩B): 0.043333333333333335
P(A∩C): 0.07833333333333334
P(A|B): 0.65
P(B|A): 0.17105263157894737
P(A)×P(B): 0.01688888888888889


In [10]:
year1 = df[df["Year_of_Study"]==1]
year4 = df[df["Year_of_Study"]==4]

intersection = len(df[(df["Year_of_Study"]==1) &
                      (df["Year_of_Study"]==4)])

print("Intersection:", intersection)

if intersection == 0:
    print("Year 1 and Year 4 are mutually exclusive events.")

Intersection: 0
Year 1 and Year 4 are mutually exclusive events.


In [11]:
mean = df["Academic_Readiness_Score"].mean()
std = df["Academic_Readiness_Score"].std()

df["Z_Score"] = (df["Academic_Readiness_Score"] - mean) / std

highest = df.loc[df["Academic_Readiness_Score"].idxmax()]
lowest = df.loc[df["Academic_Readiness_Score"].idxmin()]

print("Highest Student")
print(highest[["Student_ID","Academic_Readiness_Score","Z_Score"]])

print("Lowest Student")
print(lowest[["Student_ID","Academic_Readiness_Score","Z_Score"]])

Highest Student
Student_ID                   STU0131
Academic_Readiness_Score        98.0
Z_Score                     2.705351
Name: 130, dtype: object
Lowest Student
Student_ID                   STU0543
Academic_Readiness_Score        41.9
Z_Score                    -3.081477
Name: 542, dtype: object


In [12]:
within1 = ((df["Z_Score"]>=-1) & (df["Z_Score"]<=1)).mean()*100
within2 = ((df["Z_Score"]>=-2) & (df["Z_Score"]<=2)).mean()*100
within3 = ((df["Z_Score"]>=-3) & (df["Z_Score"]<=3)).mean()*100

print("Within 1 SD:", within1,"%")
print("Within 2 SD:", within2,"%")
print("Within 3 SD:", within3,"%")

Within 1 SD: 69.5 %
Within 2 SD: 95.16666666666667 %
Within 3 SD: 99.83333333333333 %


##Five Statistical Observations

1. Average students study 15.69 hours/week.
2. Students sleep about 7 hours/day on average.
3. Academic Readiness shows the highest variability.
4. Monthly spending has the most outliers (20).
5. The distribution of Academic Readiness is approximately normal because 69.5%, 95.17%, and 99.83% fall within 1, 2, and 3 standard deviations.